python/ray/data/dataset.py

```diff
@@ -2787,14 +2787,9 @@ def make_generator():
 
     def to_tf(
         self,
+        feature_columns: Union[str, List[str]],
+        label_columns: Union[str, List[str]],
```

Comment:
Should this be optional?

Reply:
I don't think so, because it'd complicate the function. 

`keras.Model.fit` always expects a two-tuple representing inputs and targets.

> A [tf.data](https://www.tensorflow.org/api_docs/python/tf/data) dataset. Should return a tuple of either (inputs, targets) or (inputs, targets, sample_weights).

The only time you wouldn't return a two-tuple is when you write a [training loop from scratch](https://www.tensorflow.org/guide/keras/writing_a_training_loop_from_scratch#end-to-end_example_a_gan_training_loop_from_scratch). Since it's a niche use case, I don't think we should support it in this initial PR. 

Also see #29030 for how we can support this in the future.